sinon/lib/stub.py

```diff
@@ -17,118 +17,178 @@ class SinonStub(SinonSpy):
     (4) a module
 
     All function/module will be replaced into customized or empty function/class
-    Able to be assigned special condition (on which call, with what args...etc)
+    Able to be assigned special conditions (on which call, with what args, etc.)
     """
 
     def __init__(self, obj=None, prop=None, func=None):
         super(SinonStub, self).__init__(obj, prop)
-        self.stubfunc = func if func else Wrapper.empty_function
-        super(SinonStub, self).wrap2stub(self.stubfunc)
-        # Todo: target is a dirty hack
-        self.condition = {"args":[], "kwargs":[], "action": [], "oncall":[], "target":self.obj}
-        self.cond_args = self.cond_kwargs = self.oncall = None
-
-    def __append_condition(self, func):
-        self.condition["args"].append(self.cond_args)
-        self.condition["kwargs"].append(self.cond_kwargs)
-        self.condition["oncall"].append(self.oncall)
-        self.condition["action"].append(func)
-        self.cond_args = self.cond_kwargs = self.oncall = None
+        self._stubfunc = func if func else Wrapper.empty_function
+        super(SinonStub, self).wrap2stub(self._stubfunc)
+        self._cond_args = self._cond_kwargs = self._oncall = None
```

Comment:
why you declare those variables "protected variable"?
While you declare a protected variable, I consider it will be used by other component. But I didn't find this usage in this case.

Reply:
The attributes are used in `onCall` and `withArgs` by both `SinonStub` and `SinonStubCondition`, hence they need to be _protected_. If I remove this line the code will fail.